pkg/agent/watch.go

```diff
@@ -215,28 +201,22 @@ func (e vmEvent) Format(state fmt.State, verb rune) {
 	}
 }
 
-type vmResourceTuple struct {
-	namespace  string
-	name       string
-	endpointID string
-	resource   vmResource
-	valueType  vmResourceValueType
+type kv[K any, V any] struct {
```

Comment:
This is only used in the function below, and I think it's an API we don't want to expose to the entire package. I suggest moving on the function.

Reply:
Right, already forgot about not using it more than once, so could live inside setVMMetrics